subsys/bluetooth/mesh/gen_prop_srv.c

```diff
@@ -59,7 +59,7 @@ static void store_timeout(struct k_work *work)
 
 	uint8_t user_access[CONFIG_BT_MESH_PROP_MAXCOUNT];
 
-	for (uint8_t i = 0; i < srv->property_count; ++i) {
+	for (int i = 0; i < srv->property_count; ++i) {
```

Comment:
Should `i` be uint32_t since `property_count` is uint32_t?
https://github.com/nrfconnect/sdk-nrf/blob/f4687e715c7f9581e004d265f5eedd6a66133625/include/bluetooth/mesh/gen_prop_srv.h#L173

Reply:
That was my initial thought too, but after taking to Trond we figured out that if there is not any special reason for that. We should use int as default, if it is not a special reason for it. The reason not to use uint8_t is that this can create a infinite loop, that is removed by using int.

Reply:
though is it possible to have `property_count` higher than `INT_MAX`?

Reply:
I think actual max property_count is 189 today. It has to be less than BT_MESH_PROP_MAXCOUNT.

Reply:
But that does not mater when checking with Coverity. So in some sense it would be enough with a uint8_t, but that is not something Coverity understand.

Reply:
The only thing I am not sure about is if Coverity will complain about a signed - unsigned comparison.

Reply:
signed will be converted into unsigned automatically if the equation consists of both signed and unsigned

Reply:
signed integer overflow is undefined behavior. Yes, the compiler will promote `i` to unsigned int in comparison, but if `i` is INT32_MAX, `i++` will result to undefined behavior. I would not rely on `BT_MESH_PROP_MAXCOUNT`, but rather make this code safe for every case.